viewing_party/party.py

```diff
@@ -1,23 +1,224 @@
+    popular_genre = []
+    count = 0
+
+    if watched_movies == []:
+        return None
+    for movie_data in watched_movies: 
+        popular_genre.append(movie_data["genre"])
+    for movie_data in watched_movies: 
+        if count < popular_genre.count(movie_data["genre"]):
+            count = popular_genre.count(movie_data["genre"])
+    return popular_genre[0]
+    
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
+def get_unique_watched(user_data):
+    watched_movies = user_data["watched"]
+    friends_list = user_data["friends"]
+    user_watched = []
+    dictionary_friend_movies = []
+    unique_user_movies = [] 
+
+    for movie_watched in friends_list:
+        for movie in movie_watched["watched"]:
+            dictionary_friend_movies.append(movie)
+    for movie in watched_movies:
+        user_watched.append(movie)
+        if movie not in dictionary_friend_movies:
+            unique_user_movies.append(movie)
+    return unique_user_movies
+
+
+
+def get_friends_unique_watched(user_data):
+    watched_movies = user_data["watched"]
+    friends_list = user_data["friends"]
+    unique_friend_movies = [] 
+    unique_unique = []
+
+    for movie_watched in friends_list: # {} in [{}]
+        for movie in movie_watched["watched"]: 
+            if movie not in unique_friend_movies:
+                unique_friend_movies.append(movie)
+                if movie in unique_friend_movies and movie not in watched_movies:
+                    unique_unique.append(movie)
+
+    return unique_unique
         
+
+
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+
+def get_available_recs(user_data):
+    movie_rec = []
+    watched_movies = user_data["watched"]
+    unique_movie_friends = get_friends_unique_watched(user_data)
+    user_subscriptions = user_data["subscriptions"]
+
+    for movie in unique_movie_friends:
+        if movie["host"] in user_subscriptions and movie not in watched_movies:
+            movie_rec.append(movie)
+    return movie_rec
+
+
+
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
 
+def get_new_rec_by_genre(user_data):
+    popular_genre = get_most_watched_genre(user_data)
+    unique_movie_friends = get_friends_unique_watched(user_data)
+    rec_by_genre = []
+
+    for movie in unique_movie_friends:
+        if movie["genre"] == popular_genre:
+            rec_by_genre.append(movie)
+    return rec_by_genre
+
+
+def get_rec_from_favorites(user_data):
+    user_favorites = user_data["favorites"]
+    friends_list = user_data["friends"]
+    friend_watched = []
+    rec_by_fave = []
+
+    for movie in friends_list:
+        for watched_movie in movie["watched"]:
+           friend_watched.append(watched_movie)

```

Comment:
We perform this task of condensing all the friend's movies into a single list in a few functions. This makes it a prime candidate to be pulled out to its own function for reuse!